server/src/gateway/providers/object.action.service.ts

```diff
@@ -0,0 +1,116 @@
+import { Injectable, Logger } from '@nestjs/common';
+import {
+  AddObjectData,
+  DeleteObjectData,
+  UpdateObjectData,
+} from '../gateway.dto';
+import { GwSocketWithTarget } from '../../shared/interfaces/auth/GwSocket';
+import { SlideObjectsService } from '../../modules/slide-objects/slide-objects.service';
+import { Socket } from 'socket.io';
+import { BoardPermission } from '../../enums/board.permission';
+import { ObjectAction } from '../../enums/object.action';
+
+@Injectable()
+export class ObjectActionService {
+  private readonly logger = new Logger(ObjectActionService.name);
+  constructor(private readonly slideObjectsService: SlideObjectsService) {}
+
+  async handleActionObject(
+    client: GwSocketWithTarget,
+    data: AddObjectData | UpdateObjectData | DeleteObjectData,
+    action: ObjectAction,
+  ): Promise<void> {
+    if (!this.validateUserTargetPermission(client)) return;
+
+    try {
+      switch (action) {
+        case ObjectAction.ADD:
+          return this.handleAddObject(client, data as AddObjectData);
+        case ObjectAction.UPDATE:
+          return this.handleUpdateObject(client, data as UpdateObjectData);
+        case ObjectAction.DELETE:
+          return this.handleDeleteObject(client, data as DeleteObjectData);
+        default:
+          this.emitErrorAndDisconnect(client, 'Invalid action');
```

Comment:
I think that we decided to make dedicated methods for each action, without casting the data.

Reply:
yeah, because before each method validation should be present. Moreover, they need to be wrapped into try-catch block so i think its less code now